pallets/creditcoin/src/lib.rs

```diff
@@ -581,19 +579,12 @@ pub mod pallet {
 		fn on_initialize(block_number: T::BlockNumber) -> Weight {
 			log::debug!("Cleaning up expired entries");
 
-			let unverified_task_count = match PendingTasks::<T>::remove_prefix(block_number, None) {
-				KillStorageResult::SomeRemaining(u) | KillStorageResult::AllRemoved(u) => u,
-			};
+			let unverified_task_count =
+				PendingTasks::<T>::clear_prefix(block_number, u32::MAX, None).unique;
```

Comment:
So to get things working for now I kept the behavior the same, which is to say we always remove all the entries - that's why the limit passed to the function is `u32::MAX`. That said, our current behavior isn't great.

We don't need to check the cursor because the cursor only matters for repeated calls on the same prefix, within the same block. Since we only ever clear the prefix once and it's always in `on_initialize`, the cursor will never be used.

There isn't a difference between `unique` and `backend` at the moment (the current code in substrate always returns the same value for `unique`, `backend`, and `loops`), but yeah upon rereading the docs I think you're right that `backed` is what we want.

Reply:
I think the "right" way to do this cleanup would be to limit the amount of weight we consume here, then store the returned cursor in storage, and then add code to the `on_idle` hook to consume the remaining block weight continuing to clean up if there's space left in the block. The only tricky thing is then there isn't a guarantee that the whole prefix gets cleared within this block, which would mean we would need to keep track of cleanup that didn't finish within the block, and we couldn't guarantee that storage items are actually removed when they expired.